server/src/main/java/org/elasticsearch/search/sort/FieldSortBuilder.java

```diff
@@ -368,13 +348,57 @@ public SortFieldAndFormat build(QueryShardContext context) throws IOException {
             }
             SortedNumericDVIndexFieldData numericFieldData = (SortedNumericDVIndexFieldData) fieldData;
             NumericType resolvedType = resolveNumericType(numericType);
-            field = numericFieldData.sortField(resolvedType, missing, localSortMode, nested, reverse);
+            field = numericFieldData.sortField(resolvedType, missing, localSortMode(), nested, reverse);
         } else {
-            field = fieldData.sortField(missing, localSortMode, nested, reverse);
+            field = fieldData.sortField(missing, localSortMode(), nested, reverse);
         }
         return new SortFieldAndFormat(field, fieldType.docValueFormat(null, null));
     }
 
+    @Override
+    public BucketedSort buildBucketedSort(QueryShardContext context) throws IOException {
+        MappedFieldType fieldType = context.fieldMapper(fieldName);
+        Nested nested = nested(context, fieldType);
+        if (fieldType == null) {
+            if (unmappedType != null) {
+                fieldType = context.getMapperService().unmappedFieldType(unmappedType);
+            } else {
+                throw new QueryShardException(context, "No mapping found for [" + fieldName + "] in order to sort on");
+            }
+        }
+
+        IndexFieldData<?> fieldData = context.getForField(fieldType);
+        if (fieldData instanceof IndexNumericFieldData == false
+                && (sortMode == SortMode.SUM || sortMode == SortMode.AVG || sortMode == SortMode.MEDIAN)) {
+            throw new QueryShardException(context, "we only support AVG, MEDIAN and SUM on number based fields");
+        }
+        if (numericType != null) {
+            throw new IllegalArgumentException("not yet supported");
```

Comment:
Hmmm - this looks like it should have been a nocommit for me. I don't believe you can get here now but I need to double check on it.

Reply:
GRRR! I had it confused. Yeah. I'll take a look.